pennylane/ops/qubit/hamiltonian.py

```diff
@@ -616,8 +616,16 @@ def compare(self, other):
         if isinstance(other, Hamiltonian):
             self.simplify()
             other.simplify()
-            return self._obs_data() == other._obs_data()  # pylint: disable=protected-access
 
+            for selfdata in self._obs_data():
```

Comment:
Pythonic formatting for variable names uses an underscore between individual words.
 
```suggestion
            for self_data in self._obs_data():
```